docs/design/gitattributes-eol.md

```diff
@@ -0,0 +1,804 @@
+* Support all `text` `gitattributes` features, which decides whether EOL conversion should be applied.
+    * Set. The EOL conversion is applied to the file on snapshot and update(checkout).
+    * Unset. No EOL conversion is applied.
+    * Set to string value `auto`. `jj` uses the internal heuristic to decide whether the file is binary and whether EOL conversion should be applied.
+    * Unspecified. `working-copy.eol-conversion` decides whether the file should be converted.
+    * All other cases. Act as if `text` is in the unspecified state.
+* Support all `eol` `gitattributes` features, which decides how EOL conversion should be applied.
+    * Set to string value `crlf`. Convert the file EOL to CRLF on update(checkout). Convert the file EOL to LF on snapshot. The same effect as the `input-output` `working-copy.eol-conversion` setting.
+    * Set to string value `lf`. Do not convert the file EOL on update(checkout). Convert the file EOL to LF on snapshot. The same effect as the `input` `working-copy.eol-conversion` setting.
+    * Unspecified. How EOL conversion should be applied is decided by the `working-copy.eol-conversion` setting and the `working-copy.gitattributes-default-eol` setting.
+    * All other cases. Act as if `eol` is in the unspecified state.
+    * Specifying `eol` automatically sets `text` if `text` was left unspecified.
+* Support the [`crlf` `gitattributes`](https://git-scm.com/docs/gitattributes#_backwards_compatibility_with_crlf_attribute):
+    * Set. The same as `text` is set.
+    * Unset. The same as `text` is unset.
+    * Set to string value `input`. The same as `eol` is set to `lf`.
+* How the [`GitAttributes`] type introduced in the [basic `gitattributes` design](gitattributes.md) should be initialized and used.
+* If no EOL should be applied to the file, the file should not be read when calling the EOL convert functions(`convert_eol_for_snapshot` and `convert_eol_for_update`).
+
+## State of the Feature as of v0.35.0
+
+* We have [a design doc](gitattributes.md) on how to read the `gitattributes` associated with a file.
+* We have implemented a feature similar to the git `core.autocrlf` config in [this PR](https://github.com/jj-vcs/jj/pull/6728), which implements the EOL conversion itself.
+
+## Prior work
+
+* In git, it's [`crlf_to_git`](https://github.com/git/git/blob/9a2fb147f2c61d0cab52c883e7e26f5b7948e3ed/convert.c#L503-L590) and [`crlf_to_worktree`](https://github.com/git/git/blob/9a2fb147f2c61d0cab52c883e7e26f5b7948e3ed/convert.c#L592-L627).
+* In gitoxide, it's [`gix_filter::eol::convert_to_git`](https://docs.rs/gix-filter/0.21.0/gix_filter/eol/fn.convert_to_git.html) and [`gix_filter::eol::convert_to_worktree::convert_to_worktree`](https://docs.rs/gix-filter/0.21.0/gix_filter/eol/convert_to_worktree/fn.convert_to_worktree.html).
+
+## Open questions
+
+1. Do we need more detailed description in the [Prior work](#prior-work) section?
```

Comment:
nit: did you take a look at what Mercurial/Fig does? It may be useful for another data point